routeros version not being auto-detected in latest provider version.

**Describe the bug**
We deploying a fleet of Mikrotik routers running routeros 7.13.x.   Up until this morning, new deployments using the provider were able to auto-determine the routeros version the router is running without specifying the `routeros` attribute [per the documentation](https://registry.terraform.io/providers/terraform-routeros/routeros/latest/docs#routeros-1). 

As of this morning, on provider version 1.76.2, we had to hardcode the `routeros` attribute to our mikrotik's routeros version for the plan to succeed.

As a side note, the attribute might be better something along `routeros_version` to help distinguish what it does (the provider also being called routeros can be confusing)

**To Reproduce**
Try to execute a `plan` on a MikroTik router running version 7.13.x using the latest version of the provider `1.76.2`.  

**Expected behavior**
The provider should be able to auto-determine the router's os version per the documentation.

Ok, we actually had a deployed router need an update today and ran into the RouterOS version error again.   Here is what I think is the relevant output from `terraform plan` with `TF_LOG=debug ROS_LOG_COLOR=1` enabled.

I can send the full plan log if it helps. It contains a lot of sensitive info that I'd prefer to send over privately and not on this public forum (would take forever to sanitize it)

```
 2025-02-12T13:29:04.990-0800 [DEBUG] provider.terraform-provider-routeros_v1.76.2: response body: {"architecture-name":"arm64","bad-blocks":"0","board-name":"CCR2004-16G-2S+","build-time":"Dec/14/2023 09:24:08","cpu":"ARM64","cpu-count":"4","cpu-load":"1","factory-software":"7.6","free-hdd-space":"112328704","free-memory":"3975020544","platform":"MikroTik","total-hdd-space":"134217728","total-memory":"4227858432","uptime":"5w1d21h46m2s","version":"7.13 (stable)","write-sect-since-reboot":"305450","write-sect-total":"319352"}: @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:27 @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=4e27786c-22aa-6e43-fbcc-f11556079762 tf_rpc=Configure timestamp=2025-02-12T13:29:04.989-0800
Error: -12T13:29:04.990-0800 [ERROR] provider.terraform-provider-routeros_v1.76.2: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_severity=ERROR tf_req_id=4e27786c-22aa-6e43-fbcc-f11556079762 tf_rpc=Configure @module=sdk.proto diagnostic_detail="" diagnostic_summary="RouterOS version not found" tf_proto_version=5.8 tf_provider_addr=terraform-routeros/routeros timestamp=2025-02-12T13:29:04.990-0800
Error: -12T13:29:04.991-0800 [ERROR] vertex "provider[\"registry.terraform.io/terraform-routeros/routeros\"]" error: RouterOS version not found
```

--- a/routeros/mikrotik_client.go
+++ b/routeros/mikrotik_client.go
@@ -114,7 +114,7 @@ func NewClient(ctx context.Context, d *schema.ResourceData) (interface{}, diag.D
 		panic("[NewClient] wrong transport type: " + routerUrl.Scheme)
 	}
 
-	RouterOSVersion = d.Get("routeros").(string)
+	RouterOSVersion = d.Get("routeros_version").(string)
 	if RouterOSVersion != "" {
 		ColorizedMessage(ctx, INFO, "RouterOS from env: "+RouterOSVersion)
 	}

--- a/routeros/mikrotik_client.go
+++ b/routeros/mikrotik_client.go
@@ -248,7 +248,8 @@ func GetRouterOSVersion(m interface{}) (string, diag.Diagnostics) {
 		return "", diag.Errorf("RouterOS version not found")
 	}
 
-	re := regexp.MustCompile(`^\d+\.\d+\.\d+`)
+	// d.d | d.d.d
+	re := regexp.MustCompile(`^(\d+\.){1,2}\d+`)
 
 	if !re.MatchString(version) {
 		return "", diag.Errorf("RouterOS version not found")

--- a/routeros/provider.go
+++ b/routeros/provider.go
@@ -90,7 +90,7 @@ func Provider() *schema.Provider {
 				),
 				Description: "Suppress the system object deletion warning.",
 			},
-			"routeros": {
+			"routeros_version": {
 				Type:     schema.TypeString,
 				Optional: true,
 				DefaultFunc: schema.MultiEnvDefaultFunc(

--- a/routeros/provider.go
+++ b/routeros/provider.go
@@ -60,7 +60,7 @@ func Provider() *schema.Provider {
 					[]string{"ROS_PASSWORD", "MIKROTIK_PASSWORD"},
 					nil,
 				),
-				Description: "Password for the MikroTik user.",
+				Description: "Password for the MikroTik user (env: ROS_PASSWORD | MIKROTIK_PASSWORD).",
 				Sensitive:   true,
 			},
 			"ca_certificate": {
@@ -70,7 +70,7 @@ func Provider() *schema.Provider {
 					[]string{"ROS_CA_CERTIFICATE", "MIKROTIK_CA_CERTIFICATE"},
 					nil,
 				),
-				Description: "Path to MikroTik's certificate authority file.",
+				Description: "Path to MikroTik's certificate authority file (env: ROS_CA_CERTIFICATE | MIKROTIK_CA_CERTIFICATE).",
 			},
 			"insecure": {
 				Type:     schema.TypeBool,
@@ -79,7 +79,7 @@ func Provider() *schema.Provider {
 					[]string{"ROS_INSECURE", "MIKROTIK_INSECURE"},
 					false,
 				),
-				Description: "Whether to verify the SSL certificate or not.",
+				Description: "Whether to verify the SSL certificate or not (env: ROS_INSECURE | MIKROTIK_INSECURE).",
 			},
 			"suppress_syso_del_warn": {
 				Type:     schema.TypeBool,
@@ -88,7 +88,7 @@ func Provider() *schema.Provider {
 					[]string{"ROS_SUPPRESS_SYSO_DEL_WARN"},
 					false,
 				),
-				Description: "Suppress the system object deletion warning.",
+				Description: "Suppress the system object deletion warning (env: ROS_SUPPRESS_SYSO_DEL_WARN).",
 			},
 			"routeros_version": {
 				Type:     schema.TypeString,
@@ -98,7 +98,7 @@ func Provider() *schema.Provider {
 					nil,
 				),
 				Description: "RouterOS version for which resource schemes will be adapted. The version obtained from " +
-					"MikroTik will be used if not specified.",
+					"MikroTik will be used if not specified (env: ROS_VERSION).",
 			},
 		},
 		ResourcesMap: map[string]*schema.Resource{

